soh/soh/Enhancements/randomizer/3drando/location_access/locacc_kakariko.cpp

```diff
@@ -23,14 +23,14 @@ void AreaTable_Init_Kakariko() {
                 }, {
                   //Exits
                   Entrance(RR_HYRULE_FIELD,                {[]{return true;}}),
-                  Entrance(RR_KAK_CARPENTER_BOSS_HOUSE,    {[]{return true;}}),
-                  Entrance(RR_KAK_HOUSE_OF_SKULLTULA,      {[]{return true;}}),
-                  Entrance(RR_KAK_IMPAS_HOUSE,             {[]{return true;}}),
-                  Entrance(RR_KAK_WINDMILL,                {[]{return true;}}),
-                  Entrance(RR_KAK_BAZAAR,                  {[]{return logic->IsAdult && logic->AtDay;}}),
-                  Entrance(RR_KAK_SHOOTING_GALLERY,        {[]{return logic->IsAdult && logic->AtDay;}}),
+                  Entrance(RR_KAK_CARPENTER_BOSS_HOUSE,    {[]{return (logic->CanOpenOverworldDoor(RG_BOSS_HOUSE_KEY));}}),
+                  Entrance(RR_KAK_HOUSE_OF_SKULLTULA,      {[]{return (logic->CanOpenOverworldDoor(RG_SKULLTULA_HOUSE_KEY));}}),
+                  Entrance(RR_KAK_IMPAS_HOUSE,             {[]{return (logic->CanOpenOverworldDoor(RG_IMPAS_HOUSE_KEY));}}),
+                  Entrance(RR_KAK_WINDMILL,                {[]{return (logic->CanOpenOverworldDoor(RG_WINDMILL_KEY));}}),
+                  Entrance(RR_KAK_BAZAAR,                  {[]{return logic->IsAdult && logic->AtDay && (logic->CanOpenOverworldDoor(RG_KAK_BAZAAR_KEY));}}),
+                  Entrance(RR_KAK_SHOOTING_GALLERY,        {[]{return logic->IsAdult && logic->AtDay && (logic->CanOpenOverworldDoor(RG_KAK_SHOOTING_GALLERY_KEY));}}),
                   Entrance(RR_BOTTOM_OF_THE_WELL_ENTRYWAY, {[]{return logic->DrainWell && (logic->IsChild || randoCtx->GetOption(RSK_SHUFFLE_DUNGEON_ENTRANCES).IsNot(RO_DUNGEON_ENTRANCE_SHUFFLE_OFF));}}),
-                  Entrance(RR_KAK_POTION_SHOP_FRONT,       {[]{return logic->AtDay || logic->IsChild;}}),
+                  Entrance(RR_KAK_POTION_SHOP_FRONT,       {[]{return logic->AtDay || logic->IsChild && (logic->CanOpenOverworldDoor(RG_KAK_POTION_SHOP_KEY));}}),
```

Comment:
Logic error due to operator precedence.
```suggestion
                  Entrance(RR_KAK_POTION_SHOP_FRONT,       {[]{return (logic->AtDay || logic->IsChild) && (logic->CanOpenOverworldDoor(RG_KAK_POTION_SHOP_KEY));}}),
```